### ISSUE TITLE
Installation fails due to conflicting flask version

### ISSUE THREAD

**Issue by NeolithEra** [pymacaron/pymacaron#4]
Hi, users are unable to run **_pymacaron_** due to dependency conflict with _**flask**_ package. As shown in the following full dependency graph of pymacaron, pymacaron requires _**flask**_ ，while pymacaron-core-1.0.130 requires _**flask==0.10.1**_.

According to pip’s “first found wins” installation strategy, _**flask 1.1.1**_ is the actually installed version. However, _**flask 1.1.1**_ does not satisfy _**flask==0.10.1**_.

### Dependency tree------
```
| +-pymacaron<version range:>
| | +-click<version range:>
| | +-flask<version range:>
| | +-flask-compress<version range:>
| | | +-flask<version range:>
| | +-flask-cors<version range:>
| | +-pyjwt<version range:>
| | +-pymacaron-core<version range:>
| | | +-bravado-core<version range:==5.10.0>
| | | +-flask<version range:==0.10.1>
| | | | +-itsdangerous<version range:>=0.21>
| | | | +-jinja2<version range:>=2.4>
| | | | +-werkzeug<version range:>=0.7>
| | | +-pyyaml<version range:>
| | +-pymacaron-unit<version range:>
| | | +-requests<version range:>
| | +-pytz<version range:>
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution

1.	Fix your direct dependency to be flask==0.10.1. I have checked this revision will not affect your downstream projects now.
2.	Ask your upstream project **_pymacaron-core_** to loose the version range of flask.
Which solution do you prefer, 1 or 2?
Please let me know your choice. I can submit a PR to solve this issue.

**Comment by NeolithEra** [comment]
pymacaron has two downstream projects, i.e.,  pymacaron-dynamodb-0.0.19 and pymacaron-async-0.0.38. 

As shown in the following dependency tree of these two projects, fixing your direct dependency to be flask==0.10.1 will not cause conflicts.

```
----------------------------------------------------------------------------------------
Building pymacaron-dynamodb-0.0.19
----------------------------------------------------------------------------------------
pymacaron-dynamodb<version range:>
| +-boto3<version range:>
| +-pymacaron<version range:>
| | +-click<version range:>
| | +-flask<version range:>
| | +-flask-compress<version range:>
| | | +-flask<version range:>
| | +-flask-cors<version range:>
| | +-pyjwt<version range:>
| | +-pymacaron-core<version range:>
| | | +-bravado-core<version range:==5.10.0>
| | | +-flask<version range:==0.10.1>
| | | | +-itsdangerous<version range:>=0.21>
| | | | +-jinja2<version range:>=2.4>
| | | | +-werkzeug<version range:>=0.7>
| | | +-pyyaml<version range:>
| | +-pymacaron-unit<version range:>
| | | +-requests<version range:>
| | +-pytz<version range:>
| +-pymacaron-unit<version range:>
| | +-requests<version range:>
----------------------------------------------------------------------------------------
BUILD SUCCESS
----------------------------------------------------------------------------------------
```



```
----------------------------------------------------------------------------------------
Building pymacaron-async-0.0.38
----------------------------------------------------------------------------------------
pymacaron-async<version range:>
| +-celery<version range:>
| +-pycurl<version range:>
| +-pymacaron<version range:>
| | +-click<version range:>
| | +-flask<version range:>
| | +-flask-compress<version range:>
| | | +-flask<version range:>
| | +-flask-cors<version range:>
| | +-pyjwt<version range:>
| | +-pymacaron-core<version range:>
| | | +-bravado-core<version range:==5.10.0>
| | | +-flask<version range:==0.10.1>
| | | | +-itsdangerous<version range:>=0.21>
| | | | +-jinja2<version range:>=2.4>
| | | | +-werkzeug<version range:>=0.7>
| | | +-pyyaml<version range:>
| | +-pymacaron-unit<version range:>
| | | +-requests<version range:>
| | +-pytz<version range:>
| +-pymacaron-unit<version range:>
| | +-requests<version range:>
| +-redis<version range:>
----------------------------------------------------------------------------------------
BUILD SUCCESS
----------------------------------------------------------------------------------------
```

**Comment by erwan-lemonnier** [comment]
Hi @NeolithEra! Thanks for your report! I plan on updating all dependencies to flask 1.1.1 shortly. Will let you know when it is done.

**Comment by erwan-lemonnier** [comment]
Fixed with pymacaron-core-1.0.146. Thanks for reporting it!
